Sources/Networking/Core/Upload/UploadAPIManager.swift

```diff
@@ -0,0 +1,293 @@
+            request: existingUploadTask.endpointRequest,
+            retryConfiguration: retryConfiguration
+        )
+    }
+
+    public func stateStream(for uploadTaskId: UploadTask.ID) async -> StateStream {
+        let uploadTask = await uploadTasks
+            .getValues()
+            .values
+            .first { $0.id == uploadTaskId }
+
+        return uploadTask?.stateStream ?? Empty().eraseToAnyPublisher().values
+    }
+}
+
+// MARK: - Private API
+private extension UploadAPIManager {
+    @discardableResult
+    func uploadRequest(
+        _ uploadable: Uploadable,
+        request: EndpointRequest,
+        retryConfiguration: RetryConfiguration?
+    ) async throws -> UploadTask {
+        do {
+            let urlRequest = try await prepare(request)
+
+            let sessionUploadTask = sessionUploadTask(
+                with: uploadable,
+                for: urlRequest
+            ) { [weak self] data, response, error in
+                self?.handleUploadTaskCompletion(
+                    urlRequest: urlRequest,
+                    endpointRequest: request,
+                    retryConfiguration: retryConfiguration,
+                    data: data,
+                    response: response,
+                    error: error
+                )
+            }
+
+            let uploadTask = await existingUploadTaskOrNew(
+                for: sessionUploadTask,
+                request: request,
+                uploadable: uploadable
+            )
+
+            // Store the task for future processing
+            await uploadTasks.set(value: uploadTask, for: request.id)
+            sessionUploadTask.resume()
+            return uploadTask
+        } catch {
+            throw await errorProcessors.process(error, for: request)
+        }
+    }
+
+    /// Returns any stored upload task and updates its internal URLSessionUploadTask, or creates a new one.
+    func existingUploadTaskOrNew(
+        for sessionUploadTask: URLSessionUploadTask,
+        request: EndpointRequest,
+        uploadable: Uploadable
+    ) async -> UploadTask {
+        guard var existingUploadTask = await uploadTasks.getValue(for: request.id) else {
+            return UploadTask(
+                sessionUploadTask: sessionUploadTask,
+                endpointRequest: request,
+                uploadable: uploadable
+            )
+        }
+        existingUploadTask.task = sessionUploadTask
+        return existingUploadTask
+    }
+
+    func handleUploadTaskCompletion(
+        urlRequest: URLRequest,
+        endpointRequest: EndpointRequest,
+        retryConfiguration: RetryConfiguration?,
+        data: Data?,
+        response: URLResponse?,
+        error: Error?
+    ) {
+        Task {
+            guard let uploadTask = await uploadTasks.getValue(for: endpointRequest.id) else {
+                return
+            }
+
+            var state = UploadTask.State(task: uploadTask.task)
+            if let data, let response {
```

Comment:
Since the returned task is not used, just called, the throwing statements inside are swallowed and the error is never propagated. It happened to us since we were receiving a 500 for a bad form and we never received any errors (it never hit the error processor).

Reply:
https://github.com/strvcom/ios-networking/pull/48